docs/docs/noir/concepts/data_types/integers.md

```diff
@@ -51,6 +51,55 @@ If you are using the default proving backend with Noir, both even (e.g. _u2_, _i
 
 :::
 
+
+## 128 bits Unsigned Integers
+
+The built-in structure `U128` allows you to use 128-bit unsigned integers almost like a native integer type. However, there are some differences to keep in mind:
+- You cannot cast between a native integer and `U128`
+- There is a higher performance cost when using `U128`, compared to a native type.
```

Comment:
Just a question here @guipublic don't all casts to unsigned integers have a performance cost? I was under that impression (because range checks).

Is this cost simply higher than usual? If not, we should maybe point to the performance cost at the top of the page and remove this one to avoid confusion

Reply:
There is no cast for U128, as explained in the doc, so I am not sure to understand your question.
In term of performance, converting an integer to U128 will have no cost (because U128 is the biggest), and converting to a lower bit size will not have a specific drawback because of U128, just what you should expect.

Reply:
I guess what @signorecello meant was working with native `uint`s is less performant than working with `Field`s.

This section reads like performance costs would rank as `U128` > native `uint` > `Field`.
Is that the correct way to understand it?

Reply:
Yes, the higher the bit size, the higher is the cost. But because U128 uses 2 limbs, cost of arithmetic operations are even higher, especially for multiplication.